frontend/src/metabase/visualizations/lib/settings/series.js

```diff
@@ -98,6 +100,36 @@ export function seriesSetting({
         getSeriesDefaultLinearInterpolate(vizSettings),
       readDependencies: ["display"],
     },
+    "line.style": {
+      title: t`Line style`,
+      widget: "segmentedControl",
+      props: {
+        options: [
+          { name: t`Solid`, value: "solid" },
+          { name: t`Dashed`, value: "dashed" },
+          { name: t`Dotted`, value: "dotted" },
+        ],
+      },
+      getDefault: (series, settings) => getSeriesDefaultLineStyle(settings),
+      getHidden: (single, settings) =>
+        !LINE_DISPLAY_TYPES.has(settings["display"]),
+      readDependencies: ["display"],
+    },
+    "line.size": {
+      title: t`Line size`,
+      widget: "segmentedControl",
+      props: {
+        options: [
+          { name: "S", value: 1 },
```

Comment:
I think it is safer to keep "s", "m", "l" values stored in the series settings instead of pixel width value because it potentially would be easier to change pixel value associated with the size letter

Reply:
Good call, refactored in [`3cfd24b`](https://github.com/metabase/metabase/pull/42043/commits/3cfd24ba2874e17f7e2263aed7a05903558b1761)